### PR TITLE
Extend image_name_template to allow timestamp

### LINKED ISSUE #394: Extend image_name_template to allow timestamp
**Problem/Use Case**

The current implementation for image_name_template allows four placeholders (of which FRAME_NUMBER is not described in the documentation). 

```               file_path = '%s.%s' % (filename_template.safe_substitute(
                    VIDEO_NAME=video.name,
                    SCENE_NUMBER=scene_num_format % (i + 1),
                    IMAGE_NUMBER=image_num_format % (j + 1),
                    FRAME_NUMBER=image_timecode.get_frames()), image_extension)
```
In our use case, we would like to use the image timstamp in the filename, to be able to directly match the image with the right location in the video. Specifically, we would like to use the timestamp as in milliseconds from start, so we can use integers to refer to them. 

**Proposed Implementation:**

In terms of implementation, it is pretty trivial to add this option just by adding
`FRAME_TIMESTAMP_MS = int(image_timecode.get_seconds()*1000)`. (using `FrameTimeCode.get_seconds`)

To enable users to use this, all that needs to be done is extend the documentation specifying `$FRAME_TIMESTAMP_MS` as one of the template options.

### PR DESCRIPTION
**Pull request by Veldhoen**
Fixes #394

### CODE REVIEW

**Comment by Breakthrough** [comment]
I see in your PR you've already used the `$TIMESTAMP_MS` format.  Would you be able to run `yapf` to format the resulting code so it passes the formatting checks?  Thanks!

**Review by Breakthrough — CHANGES_REQUESTED** [review]
Thanks for the PR, please address the following so this can be landed:

 - [x] Rebase ontop of the `develop` branch
 - [ ] Ensure the code is formatted correctly: `python -m yapf -i -r scenedetect tests`

**Comment by Veldhoen** [comment]
Thanks for replying so fast, even though my PR was still a draft. I'll make sure to address the issues you mentioned, hopefully later today.

**Comment by Veldhoen** [comment]
Rebased on `develop` now, and I think the formatting is in line with your standards now. 
Also added `$TIMECODE` like you suggested in the issue, and added both `$TIMECODE` and `$FRAME_NUMBER` to `test_save_images`.

Documentation still needs to be updated to explain these extensions.

File: scenedetect/scene_manager.py [hunk 1]
```diff
@@ -489,11 +489,16 @@ def save_images(scene_list: List[Tuple[FrameTimecode, FrameTimecode]],
...
-                    IMAGE_NUMBER=image_num_format % (j + 1),
-                    FRAME_NUMBER=image_timecode.get_frames()), image_extension)
+                file_path = '%s.%s' % (
+                    filename_template.safe_substitute(
+                        VIDEO_NAME=video.name,
+                        SCENE_NUMBER=scene_num_format % (i + 1),
+                        IMAGE_NUMBER=image_num_format % (j + 1),
+                        FRAME_NUMBER=image_timecode.get_frames(),
+                        TIMESTAMP_MS=int(image_timecode.get_seconds() * 1000),
+                        TIMECODE=image_timecode.get_timecode()),
```
Comment (Breakthrough) [inline scenedetect/scene_manager.py:499]: Timecodes need to be formatted specially as not all operating systems support `:` in paths.  Please replace them with `;` [as is done when splitting videos](https://github.com/Breakthrough/PySceneDetect/blob/main/scenedetect/video_splitter.py#L140-L141):

```python
TIMECODE=image_timecode.get_timecode().replace(":", ";")
```

This is most likely why `test_save_images` is failing on the Windows build currently.

File: tests/test_scene_manager.py [hunk 2]
```diff
@@ -94,7 +94,8 @@ def test_save_images(test_video_file):
     sm.add_detector(ContentDetector())
 
     image_name_glob = 'scenedetect.tempfile.*.jpg'
-    image_name_template = 'scenedetect.tempfile.$SCENE_NUMBER.$IMAGE_NUMBER'
```
Comment (Breakthrough) [inline tests/test_scene_manager.py:97]: Could you remove the whitespace present in the resulting string?  You can use string concatenation if you place the expression in brackets:
    
    image_name_template = ('scenedetect.tempfile.'
                           '$SCENE_NUMBER.$IMAGE_NUMBER.$FRAME_NUMBER.$TIMESTAMP_MS.$TIMECODE')

Thank you for updating the unit tests!

**Review by Breakthrough — CHANGES_REQUESTED** [review]
Looks like the `test_save_images` test is failing on Windows, will approve once fixed.  Thank you!

**Comment by Veldhoen** [comment]
I applied the requested changes, and extended docstring. Hope this will do!

**Review by Breakthrough — APPROVED** [review]

**Comment by Breakthrough** [comment]
Much appreciated, approved.

Thank you for the PR!

**Comment by Veldhoen** [comment]
@Breakthrough When do you plan to release 0.6.4? I created a pre-release of my fork to be able to use the functionality right away, but I'd like to use a proper release at some point, of course.

**Comment by Breakthrough** [comment]
I'm hoping to wrap up the next release by the end of this month.  If for whatever reason you need this sooner in production, I can consider pushing a pre-release to PyPI.

**Comment by Veldhoen** [comment]
If you can make it by the end of this month, that would be great. Until then we'll be fine with my ad hoc pre-release. Thanks!

**Comment by Veldhoen** [comment]
@Breakthrough do you have any updates concerning the release of 0.6.4? Would be much appreciated :)

**Comment by Breakthrough** [comment]
All feature work is complete and I'm just finishing up testing and preparing the release.  The new version should be out within a day or two.

**Pull request merged**